### PR TITLE
[AKS Fleet] Update for Fleet identity options.

### PR DESCRIPTION
**Pull request by Ealianis**
---
### Related command
`az fleet`

* Resolved issue caused by having a "none" value passed for the MSI identity dictionary.
* Resolved issues around MSI in create / patch scenario. Creates will PUT (override) the --enabled-managed-identity & --assign-identity options. Updates will PATCH said options.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by jim-minter** [comment]
Implementation lgtm @Ealianis , is it possible to add test recording coverage for the update cases?

File: src/fleet/azext_fleet/_validators.py [hunk 1]
```diff
@@ -34,12 +34,11 @@ def validate_agent_subnet_id(namespace):
     _validate_subnet_id(namespace.agent_subnet_id, "--agent-subnet-id")
 
 
-def validate_update_strategy_id(namespace):
-    if namespace.update_strategy_id is not None:
-        from msrestazure.tools import is_valid_resource_id
-        if not is_valid_resource_id(namespace.update_strategy_id):
+def validate_update_strategy_name(namespace):
+    print(namespace.update_strategy_name)
```
Comment (jim-minter) [inline src/fleet/azext_fleet/_validators.py:38]:
```suggestion
```

Comment (Ealianis) [reply src/fleet/azext_fleet/_validators.py]: Removed testing code.

File: src/fleet/azext_fleet/commands.py [hunk 2]
```diff
@@ -40,15 +40,15 @@ def load_command_table(self, _):
         g.custom_command("update", "update_fleet", supports_no_wait=True)
         g.custom_show_command("show", "show_fleet")
         g.custom_command("list", "list_fleet")
-        g.custom_command("delete", "delete_fleet", supports_no_wait=True)
+        g.custom_command("delete", "delete_fleet", supports_no_wait=True, confirmation=True)
```
Comment (kairu-ms) [inline src/fleet/azext_fleet/commands.py:43]: It's a breaking change for the current users. Please add history notes for it.

File: src/fleet/setup.py [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 # TODO: Confirm this is the right version number you want and it matches your
 # HISTORY.rst entry.
-VERSION = '0.2.7'
+VERSION = '0.2.9'
```
Comment (kairu-ms) [inline src/fleet/setup.py:19]: Please follow this [doc](https://microsoft.sharepoint.com/:w:/t/AzureCLITools237/EWG5EuACKmtNum6A3pyLakMBSaQnkjomC9XPz9ogNPS69A?e=zRopkS) for the versioning of extension.

File: src/fleet/azext_fleet/commands.py (hunk 2 above)
Comment (Ealianis) [reply src/fleet/azext_fleet/commands.py:43]: I've updated the version and notes .

File: src/fleet/setup.py (hunk 3 above)
Comment (Ealianis) [reply src/fleet/setup.py]: Updated to 0.3.0

**Comment by kairu-ms** [comment]
@Ealianis is it ready to release?

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**